crates/radix-engine-toolkit/src/transaction_types/types.rs

```diff
@@ -373,6 +375,50 @@ impl<'r> TransactionTypesReceipt<'r> {
             )
             .collect()
     }
+
+    pub fn new_nonfungibles(&self) -> Vec<NonFungibleGlobalId> {
+        let mut minted_id_list = Vec::new();
+        let mut burnt_id_list = Vec::new();
+        for (event_type, event_payload) in
+            self.commit_result.application_events.iter()
+        {
+            match event_type.0 {
+                Emitter::Method(node_id, ..) => {
+                    if node_id.entity_type().unwrap()
```

Comment:
Instead of the `node_id.entity_type().unwrap()` it might be better to use: `if let ResourceAddress::try_from(node_id.bytes())`

```rust
match ResourceAddress::try_from(node_id.as_bytes()) {
    Some(address) if !address.is_fungible() => {
        /* Some logic */
    }
    _ => {}
}
```

Generally, I would like us to avoid unwraps in all of the code paths and the above described flow allows us to avoid them.